src/validators/tuple.rs

```diff
@@ -132,6 +132,73 @@ impl BuildValidator for TuplePositionalValidator {
     }
 }
 
+#[allow(clippy::too_many_arguments)]
+fn validate_tuple_positional<'s, 'data, T: Iterator<Item = PyResult<&'data I>>, I: Input<'data> + 'data>(
+    py: Python<'data>,
+    input: &'data impl Input<'data>,
+    extra: &Extra,
+    definitions: &'data Definitions<CombinedValidator>,
+    recursion_guard: &'s mut RecursionGuard,
+    output: &mut Vec<PyObject>,
+    errors: &mut Vec<ValLineError<'data>>,
+    extra_validator: &Option<Box<CombinedValidator>>,
+    items_validators: &[CombinedValidator],
+    collection_iter: &mut T,
+    collection_len: Option<usize>,
+    expected_length: usize,
+) -> ValResult<'data, ()> {
```

Comment:
This is basically what was inside of the macro but it didn't like the `iter.map(Ok)` pattern because it couldn't infer that the Err variant was a ValResult :(